account_statement_import_file/wizard/account_statement_import.py

```diff
@@ -290,6 +284,9 @@ def _complete_stmts_vals(self, stmts_vals, journal, account_number):
         speeddict = journal._statement_line_import_speeddict()
         for st_vals in stmts_vals:
             st_vals["journal_id"] = journal.id
+            attach_vals = self._prepare_create_attachment(journal)
+            if attach_vals:
+                st_vals["attachment_ids"] = [Command.create(attach_vals)]
```

Comment:
This creates a new `ir.attachment` for each statement but we only need one `ir.attachment` and it only has to be linked do the statements.
Sure we can trust Odoo that it won't duplicate the file in the filesystem, but still, I think it should be clear that we only want to create 1 file and link it to the statements.

I have found that this issue has already been fixed in `17.0` with https://github.com/OCA/bank-statement-import/pull/692/commits/db8980dabe885f026e9aebcacc0fcded2eafaf9c, maybe you can cherry pick that commit? It would also keep the versions aligned.